services/catalog/src/simcore_service_catalog/services/access_rights.py

```diff
@@ -0,0 +1,167 @@
+from pydantic.types import PositiveInt
+
+from ..api.dependencies.director import get_director_api
+from ..db.repositories.groups import GroupsRepository
+from ..db.repositories.services import ServicesRepository
+from ..utils.versioning import as_version, is_patch_release
+
+logger = logging.getLogger(__name__)
+
+OLD_SERVICES_DATE: datetime = datetime(2020, 8, 19)
+
+
+def _is_frontend_service(service: ServiceDockerData) -> bool:
+    return "/frontend/" in service.key
+
+
+async def _is_old_service(app: FastAPI, service: ServiceDockerData) -> bool:
+    # get service build date
+    client = get_director_api(app)
+    data = await client.get(
+        f"/service_extras/{quote_plus(service.key)}/{service.version}"
+    )
+    if not data or "build_date" not in data:
+        return True
+
+    logger.debug("retrieved service extras are %s", data)
+
+    service_build_data = datetime.strptime(data["build_date"], "%Y-%m-%dT%H:%M:%SZ")
+    return service_build_data < OLD_SERVICES_DATE
+
+
+async def evaluate_default_policy(
+    app: FastAPI, service: ServiceDockerData
+) -> Tuple[Optional[PositiveInt], List[ServiceAccessRightsAtDB]]:
+    """Given a service, it returns the owner's group-id (gid) and a list of access rights following
+    default access-rights policies
+
+    - DEFAULT Access Rights policies:
+        1. All services published in osparc prior 19.08.2020 will be visible to everyone (refered as 'old service').
+        2. Services published after 19.08.2020 will be visible ONLY to his/her owner
+        3. Front-end services are have read-access to everyone
+    """
+    db_engine: Engine = app.state.engine
+
+    groups_repo = GroupsRepository(db_engine)
+
+    everyone_gid = (await groups_repo.get_everyone_group()).gid
+    owner_gid = None
+    reader_gids: List[PositiveInt] = []
+
+    if _is_frontend_service(service) or await _is_old_service(app, service):
+        logger.debug("service %s:%s is old or frontend", service.key, service.version)
+        # let's make that one available to everyone
+        reader_gids.append(everyone_gid)
+
+    # try to find the owner
+    possible_owner_email = [service.contact] + [
+        author.email for author in service.authors
+    ]
+
+    for user_email in possible_owner_email:
+        possible_gid = await groups_repo.get_user_gid_from_email(user_email)
+        if possible_gid:
+            if not owner_gid:
+                owner_gid = possible_gid
+    if not owner_gid:
+        logger.warning("service %s:%s has no owner", service.key, service.version)
+    else:
+        reader_gids.append(owner_gid)
+
+    # we add the owner with full rights, unless it's everyone
+    default_access_rights = [
+        ServiceAccessRightsAtDB(
+            key=service.key,
+            version=service.version,
+            gid=gid,
+            execute_access=True,
+            write_access=(gid == owner_gid),
+            product_name=app.state.settings.access_rights_default_product_name,
```

Comment:
default product name?

Reply:
yes, it was laready there. The default policy only applies to the default product